From 936997c756717a9e8305701648c20d043f88608e Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 9 Mar 2017 11:04:28 +0800
Subject: [PATCH 1644/5242] MLK-14299-2 video: mxsfb: setup the initial params
 for overlay fb

commit  86bd81eb2e7ed1863cce51c71d7deee26a6cb344 from
https://source.codeaurora.org/external/imx/linux-imx.git

The initial params for overlay fb should be setup during
its initialization to make it in a determined state before
using it.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit f3d3d58165f9f36af85a46256cacc274b2d0b107)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 2db0638..4dd0d83a 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -2067,6 +2067,10 @@ static void mxsfb_overlay_init(struct mxsfb_info *fbi)
 		dev_err(ofb->dev, "failed to map video mem for overlay\n");
 		goto fb_unregister;
 	}
+
+	/* setup the initial params for overlay fb */
+	overlayfb_set_par(ofb->ol_fb);
+
 	ofb->registered = 1;
 
 	return;
-- 
1.7.9.5

